### ISSUE TITLE
Tree node .parent does not appear to match expectation

### ISSUE THREAD

**Issue by najcarvalho** [tree-sitter/py-tree-sitter#61]
Hello, 

I'm using tree-sitter 0.2.0
[
% /usr/bin/pip3 freeze
tree-sitter==0.2.0
]

See this sample code (rename to parent_test.py to run):
[parent_test.py.txt](https://github.com/tree-sitter/py-tree-sitter/files/6094208/parent_test.py.txt)

I am just DFS walking the tree from the root.

Here's a small snippet of the output (beginning) it generates:

`Using source code: 
-------------------------------

	def main():
		a = 2
		
-------------------------------
Visiting node: id:4530382256, node.parent_id: None, actual_parent_id:None, b'def main():\n\t\ta = 2\n\t\t'
Node has 1 children..
Traversing child: 0
Visiting node: id:4530382384, node.parent_id: 4530382512, actual_parent_id:4530382256, b'def main():\n\t\ta = 2'
Node has 5 children..
Traversing child: 0
Visiting node: id:4530382512, node.parent_id: 4530382832, actual_parent_id:4530382384, b'def'
Node has 0 children..
Finished visiting node: id:4530382512, node.parent_id: 4530382832, actual_parent_id:4530382384, b'def'
Traversing child: 1`

----

Look at the 2nd node visited:

Visiting node: id:4530382384, node.parent_id: 4530382512, actual_parent_id:4530382256, b'def main():\n\t\ta = 2'

The node id when using node.parent is id: 4530382512
But the actual parent of this node in this traversal was id: 4530382256

So in summary, node.parent looks incorrect. 

Is this a known issue? or is node.parent supposed to mean something different than "the parent of this node in the traversal"?

**Comment by lunixbochs** [comment]
`id(node)` is just the address of the Python wrapper Node, which can be different with every call to `node.parent`. The actual Tree Sitter node ID isn't exposed yet. Maybe try `node.id` from this pull request https://github.com/tree-sitter/py-tree-sitter/pull/59

This demonstrates the issue, note that `node.parent is node.parent` prints False, showing because a new node wrapper is allocated in Python for each call to node.parent, thus `node_parent_id1 != node_parent_id2`. Note that you can't just call `id(node.parent)` twice in a row to test this, you need to store `node.parent` to a variable so the address isn't reused by the Python allocator.
```
    parent_1 = node.parent
    parent_2 = node.parent
    node_parent_id1 = get_id(parent_1)
    node_parent_id2 = get_id(parent_2)
    print(node.parent is node.parent)
```

**Comment by lunixbochs** [comment]
#59 is merged now, this should be resolved. You should use the latest commit and `node.id` rather than `id(node)`.
